# Data Analysis on Electric Vehicles
## Topic Relevance

For this project I am using the Electric Vehicle population data from data.gov.
I am choosing this data set because electric vehicles (EVs) are what the cars of the future will be based on. Right now EVs seem like a luxury item,
but 20 or so years into the future and most of these "luxury items" that are in current EVs will become a standard feature.

My current interest with this data set is to look at some of the data to see where they fit in and how feasible it is to own one today.

## Questions
1. What is the avg price of the EV's
2. What is the avg price per make
3. What is the model year range of the EVs.
4. What are the 10  Vehicles with the most range
5. What are the 10 Vehicles with the least range
6. How many different makes and models are there
7. What are the average range per make
8. What is the avg Range per model year



# Getting data into DataFrame
Importing pandas and display and the CSV file into a data frame and showing the first 5 entries ensuring the data was loaded


In [2]:
from IPython.display import display
import pandas as pd
df = pd.read_csv('Electric_Vehicle_Population_Data.csv')
df.head()

,VIN (1-10),County,City,State,Postal Code,Model Year,Make,Model,Electric Vehicle Type,Clean Alternative Fuel Vehicle (CAFV) Eligibility,Electric Range,Base MSRP,Legislative District,DOL Vehicle ID,Vehicle Location,Electric Utility,2020 Census Tract
0,1C4JJXP68P,Yakima,Yakima,WA,98901.0,2023,JEEP,WRANGLER,Plug-in Hybrid Electric Vehicle (PHEV),Not eligible due to low battery range,21.0,0.0,15.0,249905295,POINT (-120.50729 46.60464),PACIFICORP,5.307700e+10
1,1G1FY6S03L,Kitsap,Kingston,WA,98346.0,2020,CHEVROLET,BOLT EV,Battery Electric Vehicle (BEV),Clean Alternative Fuel Vehicle Eligible,259.0,0.0,23.0,141133765,POINT (-122.4977 47.79802),PUGET SOUND ENERGY INC,5.303594e+10
2,KNDCE3LG9K,King,Seattle,WA,98125.0,2019,KIA,NIRO,Battery Electric Vehicle (BEV),Clean Alternative Fuel Vehicle Eligible,239.0,0.0,46.0,3410074,POINT (-122.30253 47.72656),CITY OF SEATTLE - (WA)|CITY OF TACOMA - (WA),5.303300e+10
3,1N4AZ0CP8E,Thurston,Olympia,WA,98506.0,2014,NISSAN,LEAF,Battery Electric Vehicle (BEV),Clean Alternative Fuel Vehicle Eligible,84.0,0.0,22.0,182436474,POINT (-122.87741 47.05997),PUGET SOUND ENERGY INC,5.306701e+10
4,5YJXCAE29L,Kitsap,Silverdale,WA,98383.0,2020,TESLA,MODEL X,Battery Electric Vehicle (BEV),Clean Alternative Fuel Vehicle Eligible,293.0,0.0,23.0,1843054,POINT (-122.69275 47.65171),PUGET SOUND ENERGY INC,5.303509e+10


# General Data
Next is the top 10 most popular counties, cities, states, makes, and model and shows the different types of Electric Vehicle Type(EVT) and the types of Clean Alternative Fuel Vehicle(CAFV) Eligibility and how many there are.
This is to show some top of the vehicles and types that are in the data set

In [3]:
top_counties = df['County'].value_counts().head(10)
top_cities = df['City'].value_counts().head(10)
top_states = df['State'].value_counts().head(10)
top_makes = df['Make'].value_counts().head(10)
top_models = df['Model'].value_counts().head(10)

evt = df['Electric Vehicle Type'].value_counts().head(10)
cafv = df['Clean Alternative Fuel Vehicle (CAFV) Eligibility'].value_counts().head(10)

counties_df = top_counties.to_frame()
cities_df = top_cities.to_frame()
states_df = top_states.to_frame()
makes_df = top_makes.to_frame()
models_df = top_models.to_frame()
evt_df = evt.to_frame()
cafv_df = cafv.to_frame()



display(counties_df)
display(cities_df)
display(states_df)
display(makes_df)
display(models_df)
display(evt_df)
display(cafv_df)


,count
County,
King,120383
Snohomish,29435
Pierce,19672
Clark,14424
Thurston,8759
Kitsap,8071
Spokane,6654
Whatcom,5824
Benton,3103


,count
City,
Seattle,38046
Bellevue,11685
Vancouver,8740
Redmond,8320
Bothell,7872
Kirkland,6800
Sammamish,6754
Renton,6637
Olympia,5758


,count
State,
WA,239232
CA,119
VA,67
MD,40
TX,32
FL,23
CO,22
GA,15
NC,15


,count
Make,
TESLA,101675
CHEVROLET,17462
NISSAN,15607
FORD,12851
KIA,11747
BMW,9868
TOYOTA,9783
HYUNDAI,7639
RIVIAN,7000


,count
Model,
MODEL Y,49698
MODEL 3,36181
LEAF,13865
MODEL S,7844
BOLT EV,7380
MODEL X,6546
MUSTANG MACH-E,5215
ID.4,4976
VOLT,4708


,count
Electric Vehicle Type,
Battery Electric Vehicle (BEV),190050
Plug-in Hybrid Electric Vehicle (PHEV),49697


,count
Clean Alternative Fuel Vehicle (CAFV) Eligibility,
Eligibility unknown as battery range has not been researched,142849
Clean Alternative Fuel Vehicle Eligible,73955
Not eligible due to low battery range,22943


## Data summary
This data is showing that most of the data is coming from Washington state with over 200,000 entries being from there and California being second on that list with 119


# Question 1: What is the Average Price for an EV

In [19]:
avg_msrp = df[df['Base MSRP'] > 0]['Base MSRP'].mean()
print(f"Average MSRP of all EVs: ${avg_msrp:,.2f}")


Average MSRP of all EVs: $56,934.49


### Question 1 Analysis:
Prices of vehicles vary greatly. Some are relatively inexpensive and only cost around \\$5,000 - \\$10,000, while other vehicles can cost upwards of \\$100,000 or more. The Price of luxury vehicles tend to cost more than their regular counterpart. EVs fall into a weird category since they tend to have futuristic features but have less usability and reliability for casual everyday drives or for long distance driving. With futuristic features but a lack of reliability the avg cost of EVs from this data is around $56,934.49

# Question 2: What is the average price of the EVs per make

In [24]:
top_avg_vehicles = (
    df[['Make', 'Base MSRP']]
    .query('`Base MSRP` > 0')
    .groupby('Make', as_index=False)
    .agg({'Base MSRP': 'mean'})
    .sort_values(by='Base MSRP', ascending=False)
    .reset_index(drop=True)
)

top_avg_vehicles.index = top_avg_vehicles.index + 1
top_avg_vehicles.index.name = 'Rank'


display(top_avg_vehicles)


,Make,Base MSRP
Rank,,
1,PORSCHE,135551.515152
2,FISKER,102000.000000
3,CADILLAC,75095.000000
4,TESLA,70216.869919
5,VOLVO,56258.196721
6,BMW,52784.375000
7,CHRYSLER,39995.000000
8,MINI,36865.408805
9,SUBARU,34995.000000


### Question 2 Analysis:
The cost of a vehicle can sway the opinion and feasibility for someone to buy it. The prices of gas vehicles can vary lots and EVs are no different. Porsche's and Fisker EVs are by far the most expensive at \\$135,551 and \\$102,000 respectively with Cadillac and Tesla being the next expensive at \\$75,095 and \\$70,216 with KIA being the lowest on the list at \\$32,276. With that price being very similar to that of a new gas car, this makes EVs a more tempting choice.

# Question 3: What are the year range of the EVs
This question focuses on the range of the model's years of the EVs and the amount of EVs there are per year

In [19]:
year_range = df['Model Year'].unique()
year_range.sort()
print(f"There are {len(year_range)} year models\n The year range is {year_range[0]} - {year_range[-1]}")
print(f"Number of unique years:")
display(df['Model Year'].value_counts().sort_index(ascending = False).to_frame())

There are 21 year models
 The year range is 2000 - 2026
Number of unique years:


,count
Model Year,
2026,1
2025,14067
2024,49844
2023,60233
2022,29012
2021,20520
2020,12220
2019,10968
2018,14428


### Question 3 Analysis:
This shows that the model year range of the data is from 2000 to 2026 showing that EVs are a relatively new type of luxury vehicle that are growing in popularity. The popularity of the model years steadily increases until 2024 where the amount of vehicles drops by about 10,000. The most popular EV model years are from 2022 to 2024.

# Question 4: What Are the Top 10 EVs With the Most Range
This question gets the make model and Electric Range, removes the entries with a range of 0 and averages out the ranges and ranks them from best to worst

In [62]:
top_avg_vehicles = (
    df[['Make', 'Model', 'Electric Range']]
    .query('`Electric Range` > 0')
    .groupby(['Make', 'Model'], as_index=False)
    .agg({'Electric Range': 'mean'})
    .sort_values(by='Electric Range', ascending=False)
    .head(10)
    .reset_index(drop=True)
)

top_avg_vehicles.index = top_avg_vehicles.index + 1
top_avg_vehicles.index.name = 'Rank'


display(top_avg_vehicles)


,Make,Model,Electric Range
Rank,,,
1,PORSCHE,MACAN,304.049383
2,TESLA,MODEL Y,291.000000
3,HYUNDAI,KONA,258.000000
4,CHEVROLET,BOLT EV,244.747038
5,TESLA,MODEL X,241.066933
6,TESLA,MODEL 3,238.028734
7,TESLA,ROADSTER,234.893617
8,JAGUAR,I-PACE,234.000000
9,POLESTAR,PS2,233.000000


### Question 4 Analysis:
One of the biggest draw backs of EVs are their range. gas vehicles can have a range of up to and beyond 500 miles while Evs tend to have much less. The highest range of an EV recorded on the dataset was 304 miles, which is still a long distance but not as long as most gas vehicles


# Question 5: What Are the Worst 10 EV Ranges
This question does the same as the previous but get the 10 worst instead of the best

In [5]:
top_avg_vehicles = (
    df[['Make', 'Model', 'Electric Range', 'Clean Alternative Fuel Vehicle (CAFV) Eligibility']]
    .query('`Electric Range` > 0 and `Clean Alternative Fuel Vehicle (CAFV) Eligibility` == "Clean Alternative Fuel Vehicle Eligible"')
    .groupby(['Make', 'Model'], as_index=False)
    .agg({'Electric Range': 'mean'})
    .sort_values(by='Electric Range', ascending=False)
    .tail(10)
    .reset_index(drop=True)
)

top_avg_vehicles.index = top_avg_vehicles.index + 1
top_avg_vehicles.index.name = 'Rank'


display(top_avg_vehicles)



,Make,Model,Electric Range
Rank,,,
1,LEXUS,TX,33.000000
2,FISKER,KARMA,33.000000
3,HYUNDAI,TUCSON,32.981419
4,VOLVO,XC90,32.636364
5,CHRYSLER,PACIFICA,32.171466
6,DODGE,HORNET,32.000000
7,KIA,SORENTO,31.559364
8,BMW,XM,31.000000
9,CADILLAC,CT6,31.000000


### Question 5 Analysis:
At the lowest end of the spectrum of range, we see that these EVs don't get much range at all.
From the data set the Electric Ranges go from 30 to 33 With the top 3 worst vehicles being:

1. Hyundai Santa Fe - 30
2. Cadillac CT6 - 31
3. BMW MX - 31


# Question 6: How many makes and models are there
This question will look at the total number of makes and models and the most common makes and models

In [59]:
print(f"Total amount of Makes: {df['Make'].unique().shape[0]}")
print(f"Total amount of Models: {df['Model'].unique().shape[0]}")
model_counts = (
    df[['Make', 'Model']]
    .value_counts()
    .reset_index(name='Count')
    .sort_values(by='Count', ascending=False)
)
model_counts.index = model_counts.index + 1
model_counts.index.name = 'Rank'
display(df['Make'].value_counts().head(10).to_frame())
display(model_counts.head(10))


Total amount of Makes: 46
Total amount of Models: 173


,count
Make,
TESLA,101675
CHEVROLET,17462
NISSAN,15607
FORD,12851
KIA,11747
BMW,9868
TOYOTA,9783
HYUNDAI,7639
RIVIAN,7000


,Make,Model,Count
Rank,,,
1,TESLA,MODEL Y,49698
2,TESLA,MODEL 3,36181
3,NISSAN,LEAF,13865
4,TESLA,MODEL S,7844
5,CHEVROLET,BOLT EV,7380
6,TESLA,MODEL X,6546
7,FORD,MUSTANG MACH-E,5215
8,VOLKSWAGEN,ID.4,4976
9,CHEVROLET,VOLT,4708


### Question 6 Analysis:
From the table we see that there are a total of 46 total car makes and 713 total models

The 3 most populat Makes are:
1. Tesla - 101,675
2. Chevrolet - 17,462
3. Nissan - 15,607

The 3 most popular Models are:
1. Tesla Model Y - 49,698
2. Tesla Model 3 - 36,181
3. Nissan Leaf - 13,865

# Question 7: What is the Average Range Per Make
This question will get the average Electric Range of each Make and shows the top 10 best range per Make

In [64]:
top_avg_vehicles = (
    df[['Make', 'Electric Range']]
    .query('`Electric Range` > 0')
    .groupby('Make', as_index=False)
    .agg({'Electric Range': 'mean'})
    .sort_values(by='Electric Range', ascending=False)
    .head(10)
    .reset_index(drop=True)
)

top_avg_vehicles.index = top_avg_vehicles.index + 1
top_avg_vehicles.index.name = 'Rank'


display(top_avg_vehicles)

,Make,Electric Range
Rank,,
1,TESLA,241.006721
2,JAGUAR,234.000000
3,POLESTAR,233.000000
4,CHEVROLET,147.153961
5,VOLKSWAGEN,107.604478
6,NISSAN,105.479669
7,WHEEGO ELECTRIC CARS,100.000000
8,TH!NK,100.000000
9,PORSCHE,95.130334


### Question 7 Analysis:
When buying a car, range is one of the most important factors since the longer the range the less you have to spend on fuel. With EVs generally having less range, it's important to know which EVs have enough range to support your daily activities. The table shows that Tesla, Jaguar, and Polestar have the highest Electric Range and have almost 100 more miles of range then the next choice.

# Question 8: What is the Average Range for each year

In [12]:
range_by_year = (
    df[['Model Year', 'Electric Range']]
    .query('`Electric Range` > 0')
    .groupby('Model Year', as_index=False)
    .agg(
        Vehicle_Count=('Electric Range', 'count'),
        Electric_Range=('Electric Range', 'mean')
    )
    .sort_values(by='Model Year', ascending=False)
    .reset_index(drop=True)
)

range_by_year.index = range_by_year.index + 1
range_by_year.index.name = 'Rank'

display(range_by_year)

,Model Year,Vehicle_Count,Electric_Range
Rank,,,
1,2025,2984,36.356568
2,2024,10521,33.985838
3,2023,7841,30.090677
4,2022,4675,28.987380
5,2021,4837,48.049618
6,2020,12146,238.709369
7,2019,10964,176.427308
8,2018,14428,156.834835
9,2017,8693,117.208329


### Question 8 Analysis:
Over time product tend to get better and companies try to innovate. With range being one of the limiting factors in EVs, it makes sense that companies would try to innovate new ways to get their cars to travel for longer distances without having to charge. From the table we can see that there has been some improvement with range with more years having a higher range number
The tables shows that the average range of EVs have increased overtime with a bit of a bump from 2008-2010 and a dip from 2021-present.

# Summary
My choice for picking this data set came from my interest in electric vehicles. Electric vehicles have been around for decades, but until Tesla, they were never super popular ,reliable, or practical. Early EVs were often seen as niche options, praised mostly for being quiet and eco-friendly but lacking the luxury, range, and performance of gas powered cars. Tesla flipped that perception by delivering high-tech and high performance vehicles that felt like they belonged in the future.

Today Tesla dominates the EV market making up nearly half of all vehicles within the dataset. More importantly, they lead in electric range, averaging 241 miles per charge. Tesla’s emphasis on cutting-edge features like Autopilot, minimalist interiors, and software-driven updates has raised the bar for what drivers expect from modern Electric vehicles and luxury vehicles.

Other luxury EVs like Rivian, Lucid, and Mercedes are entering the space with their own innovations. Features like self-driving capabilities, massive infotainment displays are quickly becoming normal and will likely be the standard in vehicles of the future. It's not about going electric, it’s about upgrading the entire driving experience and I have an interest in seeing how cars will evolve.

The data here shows different aspects of how EVs are performing and how they have improved overtime. The previous question shows over time EVs have overall improved in their ability to travel distances and are continuing to improve.